lib/swoosh/gallery/layout.ex

```diff
@@ -26,12 +27,14 @@ defmodule Swoosh.Gallery.Layout do
   """
   @spec render(atom, keyword) :: binary
   def render(gallery, assigns \\ []) do
+    previews = Gallery.eval_details(gallery.previews)
```

Comment:
where we are calling `render/2` without the details being set? Maybe we can fix that there, so the function doesn't need to be public 🤔

Reply:
Since gallery is a module and `.previews` here is `previews/0`, there is no way to update the previews list without turning the Gallery into a map. Do we want to do that? 🤔

Reply:
We could create a `Gallery.get/0` function and use that instead of `Gallery.previews/0` and `Gallery.groups/0`, and make them private.

Reply:
sounds good, and we could call `Gallery.get/0` on the current `Layout.render/2` call sites, so this module doesn't need to know to build the gallery data structure